includes/reader-revenue/woocommerce/class-woocommerce-connection.php

```diff
@@ -368,7 +368,7 @@ public static function should_sync_order( $order ) {
 	 * @param bool|string $payment_page_url Payment page URL. If not provided, checkout URL will be used.
 	 */
 	public static function sync_reader_from_order( $order, $verify_created_via = true, $payment_page_url = false ) {
-		if ( ! self::can_sync_customers() ) {
+		if ( ! $order || ! self::can_sync_customers() ) {
```

Comment:
isn't it safer to check if $order is an instance of WC_Order?

Reply:
Yes, you're right. ee65f29c updates this and also moves the check to the existing `should_sync_order()` validator method.